File: 프로그래머스/unrated/181846. 두 수의 합/두 수의 합.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

string solution(string a, string b) {
    string answer = "";
    if(a.size() > b.size()){
        int s = b.size();
        for(int i=0;i<a.size() - s; i++){
            b = "0" + b;
        }
    } else{
        int s = a.size();
        for(int i=0;i<b.size() - s; i++){
            a = "0" + a;
        }
    }
    int carry = 0;
    for(int i=a.size()-1; i>= 0; i--){
        int aa = a[i] - '0';
        int bb = b[i] - '0';
        if(aa + bb + carry >= 10){
            answer = to_string( (aa + bb + carry) % 10) + answer;
            carry = 1;
        } else{
            answer = to_string( (aa + bb + carry) % 10) + answer;
            carry = 0;
        }
        
    }
    if(carry == 1)
        answer = "1" + answer;
    return answer;
}

File: 프로그래머스/unrated/181846. 두 수의 합/두 수의 합.java
class Solution {
    public String solution(String a, String b) {
        String answer = "";
        if(a.length() > b.length()){
            int size = b.length();
            for(int i=0;i<a.length() - size; i++){
                b = "0" + b;    
            }
        } else{
            int size = a.length();
            for(int i=0;i<b.length() - size; i++){
                a = "0" + a;    
            }
        }
        // System.out.println(a);System.out.println(b);
        int carry = 0;
        for(int i=a.length()-1; i>=0;i--){
            int aa = Integer.valueOf(a.charAt(i) - '0');
            int bb = Integer.valueOf(b.charAt(i) - '0');
            
            if(aa + bb + carry >= 10){
                answer = String.valueOf( (aa + bb + carry ) % 10) + answer;
                carry = 1;
            }
            else{
                answer = String.valueOf( (aa + bb + carry ) % 10) + answer;
                carry = 0;
            }            
        }
        if(carry == 1)
            answer = "1" + answer;
        return answer;
    }
}